optimize/apis-tools/optimize-api/tutorial.md

```diff
@@ -0,0 +1,162 @@
+const optimizeApiUrl = process.env.OPTIMIZE_BASE_URL;
+```
+
+4. Script the API endpoint to list the dashboard IDs of a particular collection:
+
+```
+const url = `${optimizeApiUrl}/api/public/dashboard?collectionId=${collectionId}`;
+```
+
+5. Configure your GET request to the appropriate endpoint, including an authorization header based on the previously acquired `accessToken`:
+
+```
+  const options = {
+    method: "GET",
+    url,
+    headers: {
+      Accept: "application/json",
+      Authorization: `Bearer ${accessToken}`
+    }
+  };
+```
+
+6. Call the collection's endpoint, process the results from the API call, emit the dashboard IDs to output, and emit an error message from the server if necessary:
+
+```
+  try {
+    const response = await axios(options);
+    const results = response.data;
+
+    results.forEach(x => console.log(`ID: ${x.id}`));
+  } catch (error) {
+    // Emit an error from the server.
+    console.error(error.message);
+  }
+```
+
+7. In your terminal, run `node cli.js optimize list` for a list of your existing dashboard IDs.
+
+:::note
+This `list` command is connected to the `listDashboards` function at the bottom of the `optimize.js` file, and executed by the `cli.js` file. While we will view dashboard IDs and delete a dashboard in this tutorial, you may add additional arguments depending on the API calls you would like to make.
+:::
+
+If you have any existing dashboards, the `ID: ${x.id}` will now output. If you have an invalid API name or action name, or no arguments provided, or improper/insufficient credentials configured, an error message will output as outlined in the `cli.js` file.
+
+## DELETE a dashboard
+
+To delete a dashboard, take the following steps:
+
+1. Outline your function, similar to the steps above:
+
+```
+async function deleteDashboard([dashboardId]) {
+console.log(`deleting dashboard ${dashboardId}`);
+
+const optimizeAudience = process.env.OPTIMIZE_AUDIENCE;
+const accessToken = await getAccessToken("components", optimizeAudience);
+const optimizeApiUrl = process.env.OPTIMIZE_API_URL;
+
+const url = `${optimizeApiUrl}/public/dashboard/${dashboardId}`;
+```
+
+2. Configure the API call using the DELETE method:
+
+```
+  const options = {
+    method: "DELETE",
+    url,
+    headers: {
+      Accept: "application/json",
+      Authorization: `Bearer ${accessToken}`
+    }
+  };
+```
+
+3. Process the results from the API call. For example:
+
+```
+  try {
+    // Call the delete endpoint.
+    const response = await axios(options);
+
+    // Process the results from the API call.
+    if (response.status === 204) {
+      console.log(`Dashboard ${clientId} was deleted!`);
+    } else {
+      // Emit an unexpected error message.
+      console.error("Unable to delete dashboard!");
+    }
+  } catch (error) {
+    // Emit an error from the server.
+    console.error(error.message);
+  }
+}
+```
+
+4. In your terminal, run `node cli.js optimize delete` to delete your dashboard.
```

Comment:
non-blocking question:

again, should we show example output? (In this case it's just the text "dashboard was deleted!" so I think it's less valuable than showing example output on the first method. But I'm still asking the question.)

Reply:
Added in latest commit 👍